pkg/search/parser.go

```diff
@@ -108,16 +110,12 @@ func parsePair(pair string, allowEmpty bool) (key string, values string, valid b
 }
 
 func queryFromFieldValues(field string, values []string, highlight bool) *v1.Query {
-	queryLength := len(values)
-	if queryLength > MaxQueryParameters {
-		queryLength = MaxQueryParameters
-	}
-	queries := make([]*v1.Query, 0, queryLength)
-	for idx, value := range values {
-		if idx == MaxQueryParameters {
-			log.Warnf("the query contained %d parameters and was truncated to %d parameters", len(values), MaxQueryParameters)
-			break
-		}
+	// A SQL query can have no more than 65535 parameters.
+	if len(values) > MaxQueryParameters {
+		utils.Should(errors.Errorf("too many parameters %d for a query.  No more than %d parameters allowed in single query", len(values), MaxQueryParameters))
```

Comment:
Can we just use the UNEXPECTED flag? If we use should then this will blow up on scale tests